### PR TITLE
comp_roles: check product_name only for 1029u

### PR DESCRIPTION
**Pull request by asyedham**

### CODE REVIEW

**Comment by asyedham** [comment]
rerun tests

File: composable_prepare_nic_configs.yml [hunk 1]
```diff
@@ -26,16 +26,17 @@
...
+            with_together:
+              - "{{ machine_types }}"
+              - "{{ host_list.results }}"
 
-          - name: set_fact for rhel8_interfaces
+          - name: set fact rhel8_interfaces
             set_fact:
-              rhel8_interfaces: "{{  rhel8_interfaces|default([]) + [((item.stdout == 'TN10RT') | ternary('rhel8_interfaces_tn10rt', (item.stdout == 'TRTP') | ternary('rhel8_interfaces_tn10rt','rhel8_interfaces')))] }}"
-            with_items: "{{ product_name.results }}"
+              rhel8_interfaces: "{{  rhel8_interfaces|default([]) + [((item.changed == False) | ternary('rhel8_interfaces', ( item.changed == true and item.stdout == 'TN10RT') | ternary('rhel8_interfaces_tn10rt', 'rhel8_interfaces_trtp')))] }}"
```
Comment (masco) [inline composable_prepare_nic_configs.yml:37]: ( item.changed == true and item.stdout == 'TN10RT') | ternary('rhel8_interfaces_tn10rt', 'rhel8_interfaces_trtp') this part will be executed when item.changed == true right? if so, do we need to check for item.changed == true again?

Comment (asyedham) [reply composable_prepare_nic_configs.yml:37]: yes, it will be executed when item.changed == true i.e only in case 1029u's. No need of checking it again

Comment (masco) [reply composable_prepare_nic_configs.yml:37]: cool. can you remove the redundant check?

**Review by masco — CHANGES_REQUESTED** [review]

File: composable_prepare_nic_configs.yml (hunk 1 above)
Comment (asyedham) [reply composable_prepare_nic_configs.yml:37]: All the checks are needed,the condition works as follows
```
item.changed == False (i.e if machine type is not 1029u)the 
rhel8_interfaces: 'rhel8_interfaces'

If machine type 1029u and product name is tn10rt
rhel8_interfaces: 'rhel8_interfaces_tn10rt'

If machine type is 1029u and product name is trtp
rhel8: 'rhel8_interfaces_trtp'```

https://github.com/redhat-performance/jetpack/blob/master/group_vars/all.yml#L44

Comment (masco) [reply composable_prepare_nic_configs.yml:37]: I feel item.changed == true is redundant. can you check once again

Comment (asyedham) [reply composable_prepare_nic_configs.yml:37]: It is needed as without item.changed == true ansible will throw the following error
```
TASK [set fact rhel8_interfaces] *************************************************************************************************************************************************************************************************************
ok: [localhost] => (item={u'stderr_lines': [], u'ansible_loop_var': u'item', u'end': u'2020-12-08 10:47:28.557792', u'stdout': u'TRTP', u'item': [u'1029u', {u'stderr_lines': [], u'ansible_loop_var': u'item', u'end': u'2020-12-08 10:41:59.172400', u'stdout': u'f08-h06-000-1029u.rdu2.scalelab.redhat.com', u'item': u'0', u'changed': True, u'rc': 0, u'failed': False, u'cmd': u'echo "f08-h06-000-1029u.rdu2.scalelab.redhat.com"\n', u'stderr': u'', u'delta': u'0:00:00.004200', u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'strip_empty_ends': True, u'_raw_params': u'echo "f08-h06-000-1029u.rdu2.scalelab.redhat.com"\n', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin_add_newline': True, u'stdin': None}}, u'stdout_lines': [u'f08-h06-000-1029u.rdu2.scalelab.redhat.com'], u'start': u'2020-12-08 10:41:59.168200'}], u'changed': True, u'rc': 0, u'failed': False, u'cmd': u'dmidecode -t system | grep "Product Name" | cut -d \'-\' -f3\n', u'stderr': u'', u'delta': u'0:00:00.003478', u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'strip_empty_ends': True, u'_raw_params': u'dmidecode -t system | grep "Product Name" | cut -d \'-\' -f3\n', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin_add_newline': True, u'stdin': None}}, u'stdout_lines': [u'TRTP'], u'start': u'2020-12-08 10:47:28.554314'})
ok: [localhost] => (item={u'stderr_lines': [], u'ansible_loop_var': u'item', u'end': u'2020-12-08 10:47:25.283336', u'stdout': u'TRTP', u'item': [u'1029u', {u'stderr_lines': [], u'ansible_loop_var': u'item', u'end': u'2020-12-08 10:41:59.373139', u'stdout': u'f08-h07-000-1029u.rdu2.scalelab.redhat.com', u'item': u'1', u'changed': True, u'rc': 0, u'failed': False, u'cmd': u'echo "f08-h07-000-1029u.rdu2.scalelab.redhat.com"\n', u'stderr': u'', u'delta': u'0:00:00.003956', u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'strip_empty_ends': True, u'_raw_params': u'echo "f08-h07-000-1029u.rdu2.scalelab.redhat.com"\n', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin_add_newline': True, u'stdin': None}}, u'stdout_lines': [u'f08-h07-000-1029u.rdu2.scalelab.redhat.com'], u'start': u'2020-12-08 10:41:59.369183'}], u'changed': True, u'rc': 0, u'failed': False, u'cmd': u'dmidecode -t system | grep "Product Name" | cut -d \'-\' -f3\n', u'stderr': u'', u'delta': u'0:00:00.003594', u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'strip_empty_ends': True, u'_raw_params': u'dmidecode -t system | grep "Product Name" | cut -d \'-\' -f3\n', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin_add_newline': True, u'stdin': None}}, u'stdout_lines': [u'TRTP'], u'start': u'2020-12-08 10:47:25.279742'})
ok: [localhost] => (item={u'stderr_lines': [], u'ansible_loop_var': u'item', u'end': u'2020-12-08 10:47:29.921980', u'stdout': u'TRTP', u'item': [u'1029u', {u'stderr_lines': [], u'ansible_loop_var': u'item', u'end': u'2020-12-08 10:41:59.573113', u'stdout': u'f08-h08-000-1029u.rdu2.scalelab.redhat.com', u'item': u'2', u'changed': True, u'rc': 0, u'failed': False, u'cmd': u'echo "f08-h08-000-1029u.rdu2.scalelab.redhat.com"\n', u'stderr': u'', u'delta': u'0:00:00.004053', u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'strip_empty_ends': True, u'_raw_params': u'echo "f08-h08-000-1029u.rdu2.scalelab.redhat.com"\n', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin_add_newline': True, u'stdin': None}}, u'stdout_lines': [u'f08-h08-000-1029u.rdu2.scalelab.redhat.com'], u'start': u'2020-12-08 10:41:59.569060'}], u'changed': True, u'rc': 0, u'failed': False, u'cmd': u'dmidecode -t system | grep "Product Name" | cut -d \'-\' -f3\n', u'stderr': u'', u'delta': u'0:00:00.005199', u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'strip_empty_ends': True, u'_raw_params': u'dmidecode -t system | grep "Product Name" | cut -d \'-\' -f3\n', u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin_add_newline': True, u'stdin': None}}, u'stdout_lines': [u'TRTP'], u'start': u'2020-12-08 10:47:29.916781'})
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'stdout'\n\nThe error appears to be in '/root/jetpack/composable_prepare_nic_configs.yml': line 35, column 13, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n          - name: set fact rhel8_interfaces\n            ^ here\n"}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
localhost                  : ok=78   changed=24   unreachable=0    failed=1    skipped=80   rescued=0    ignored=6   
undercloud                 : ok=13   changed=6    unreachable=0    failed=0    skipped=10   rescued=0    ignored=1   
```

Comment (masco) [reply composable_prepare_nic_configs.yml:37]: strange. maybe an ansible bug. ok, we will keep it as it is for now.

**Review by masco — APPROVED** [review]

**Pull request merged**